include/aws/s3/private/s3_meta_request_impl.h

```diff
@@ -56,15 +58,19 @@ struct aws_s3_meta_request_vtable {
      * progress, false if there is not. */
     bool (*update)(struct aws_s3_meta_request *meta_request, uint32_t flags, struct aws_s3_request **out_request);
 
+    /* Run vtable->prepare_request_async() on the meta-request's event loop.
+     * We do this because body streaming is slow, and we don't want it on our networking threads.
+     * The callback may fire on any thread (an async sub-step may run on another thread). */
     void (*schedule_prepare_request)(
         struct aws_s3_meta_request *meta_request,
         struct aws_s3_request *request,
         aws_s3_meta_request_prepare_request_callback_fn *callback,
         void *user_data);
 
-    /* Given a request, prepare it for sending (ie: creating the correct HTTP message, reading from a stream (if
-     * necessary), signing it, computing hashes, etc.) */
-    int (*prepare_request)(struct aws_s3_meta_request *meta_request, struct aws_s3_request *request);
+    /* Given a request, asynchronously prepare it for sending
+     * (creating the correct HTTP message, reading from a stream (if necessary), signing it, computing hashes, etc.).
```

Comment:
Already exist, but still: This vtable function doesn't sign the request, after the function returns. I guess remove it will be more clear?

Reply:
Ooh, thanks, good point
I guess I still didn't understand the full flow after spending like a week editing this code 😅